### PR TITLE
cura: 4.4.0 -> 4.5.0

### PR DESCRIPTION
**Pull request by baracoder**
###### Motivation for this change

New release of ultimaker cura: 4.5.0
Based on #81746 

I have created one commit which also updates the dependencies, since it looks like they have to be updated in sync. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/cura/default.nix [hunk 1]
```diff
@@ -2,20 +2,20 @@
...
   pname = "cura";
-  version = "4.4.0";
+  version = "4.5.0";
 
   src = fetchFromGitHub {
     owner = "Ultimaker";
     repo = "Cura";
-    rev = "v${version}";
-    sha256 = "131n36qhdfky584wr3zv73ckjjprwaqb5fih8yln2syf8b7ziwlz";
+    rev = "${version}";
```
Comment (jonringer) [inline pkgs/applications/misc/cura/default.nix]:
```suggestion
    rev = version;
```

**Review by jonringer — COMMENTED** [review]
not part of your PR, but cura isn't properly wrapped, it's picking up other python packages:
```
stat("/nix/store/6x1q7dfyqva4w3fh524ci9czx2i8bj4z-python3.8-sip-4.19.18/lib/python3.8/site-packages/sip.so", {st_mode=S_IFREG|0555, st_size=143416, ...}) = 0
openat(AT_FDCWD, "/nix/store/6x1q7dfyqva4w3fh524ci9czx2i8bj4z-python3.8-sip-4.19.18/lib/python3.8/site-packages/sip.so", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0`X\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0555, st_size=143416, ...}) = 0
mmap(NULL, 146064, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fcb219c0000
mmap(0x7fcb219c5000, 86016, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x5000) = 0x7fcb219c5000
mmap(0x7fcb219da000, 28672, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1a000) = 0x7fcb219da000
mmap(0x7fcb219e1000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x20000) = 0x7fcb219e1000
close(3)                                = 0
mprotect(0x7fcb219e1000, 4096, PROT_READ) = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x20} ---
+++ killed by SIGSEGV (core dumped) +++
Segmentation fault (core dumped)
```
you can see that the python38 packages are being imported

**Comment by baracoder** [comment]
> not part of your PR, but cura isn't properly wrapped, it's picking up other python packages:
> ...

Did this also occur with 4.4.0?

I might have missed `libsavitar` dependency which is updated in PR #75130 in lockstep.
I will update this PR later on.

**Comment by jonringer** [comment]
It's mostly because the nix-shell is loading shellhooks when I'm reviewing, which for python packages, puts them on the PYTHONPATH. And in this case, it's picking up packages for the wrong interpreter, and causing issues.

However, this is similar to how a user has their environment. So the packages could be leaking.

**Comment by nh2** [comment]
> not part of your PR, but cura isn't properly wrapped, it's picking up other python packages:

@jonringer Right, the problem with that is (you probably know that, just writing it down extra clearly for myself and others):

`nix-review` builds for this PR all packages afffected by the update (which is Python 3.7 packages but also 3.8):

```
6 package built:
cura curaengine python37Packages.libarcus python37Packages.uranium python38Packages.libarcus python38Packages.uranium
```

And then `nix-review` puts them all on `PYTHONPATH`:

```
[nix-shell:~/.cache/nixpkgs-review/pr-82196]$ echo $PYTHONPATH
/nix/store/9cgd8d2wzh0y1d8r7ii27yqah7mxiqbw-python3.8-libarcus-4.5.0/lib/python3.8/site-packages:...
```

And then cura/Python picks them up and crashes.

Is this an issue with Cura's wrapping though? It happens even with plain `python3.7`:

```
[nix-shell:~/.cache/nixpkgs-review/pr-82196]$ python3.7
Python 3.7.6 (default, Dec 18 2019, 19:23:55) 
[GCC 9.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.path
['', '/nix/store/9cgd8d2wzh0y1d8r7ii27yqah7mxiqbw-python3.8-libarcus-4.5.0/lib/python3.8/site-packages', '/nix/store/rs7ah1yr4x4gyd0gf48hykp6lcg9mc9h-python3.8-sip-4.19.18/lib/python3.8/site-packages', '/nix/store/k82p3h05vbca3k5f1in5iavgbqxq03bm-python3-3.8.2/lib/python3.8/site-packages', '/nix/store/ln2v42znyh3n0cck1b3gjap7q4f03y7v-python3.8-uranium-4.5.0/lib/python3.8/site-packages', '/nix/store/d4xmm1xrlczhibhqimwfhg7wzc114132-python3.8-dbus-python-1.2.12/lib/python3.8/site-packages', '/nix/store/6fral947x6s9arrhk81cxy2fhvj8fir3-python3.8-PyQt5.sip-4.19.18/lib/python3.8/site-packages', '/nix/store/gscy7l3kfbcqlmbv2v5w7p7f3mbpyyvs-python3.8-pyqt-5.13.0/lib/python3.8/site-packages', '/nix/store/qrdy78g9kvlccjq7w81b84206dw75y1x-python3.8-numpy-1.18.1/lib/python3.8/site-packages', '/nix/store/x4hdmagmwd33m9c7ajs5abqc07mr7976-python3.8-scipy-1.4.1/lib/python3.8/site-packages', '/nix/store/rbcz5kgf39xbws64vqhfzxfh1yjh7hhj-python3.8-Shapely-1.6.4.post2/lib/python3.8/site-packages', '/nix/store/xb1mp0f16lv23rwhfdc1665v5r20jfmm-python3-3.7.6/lib/python37.zip', '/nix/store/xb1mp0f16lv23rwhfdc1665v5r20jfmm-python3-3.7.6/lib/python3.7', '/nix/store/xb1mp0f16lv23rwhfdc1665v5r20jfmm-python3-3.7.6/lib/python3.7/lib-dynload', '/nix/store/xb1mp0f16lv23rwhfdc1665v5r20jfmm-python3-3.7.6/lib/python3.7/site-packages']

>>> import Arcus
Segmentation fault (core dumped)
```

Isn't this a fundamental problem with `nix-review` putting everything into 1 shell?

File: pkgs/applications/misc/cura/default.nix (hunk 1 above)
Comment (nh2) [reply pkgs/applications/misc/cura/default.nix]: Done.

**Review by nh2 — APPROVED** [review]
> I might have missed `libsavitar` dependency which is updated in PR #75130 in lockstep.
> I will update this PR later on.

I've taken the liberty to force-push that improvement to this PR, and also clean up the git history to not include a merge commit.

Works fine. Will merge after next ofborg build.

Thanks @baracoder!

**Review by jonringer — APPROVED** [review]
LGTM
cura launches fine

```
[10 built, 15 copied (18.5 MiB), 3.7 MiB DL]
https://github.com/NixOS/nixpkgs/pull/82196
8 package built:
cura curaengine python37Packages.libarcus python37Packages.libsavitar python37Packages.uranium python38Packages.libarcus python38Packages.libsavitar python38Packages.uranium
```

wrapping of the executable can be done in another PR. This is still an improvement

**Comment by baracoder** [comment]
Thank you @jonringer and @nh2 
Sorry I did not find the time to update this PR like promised.

**Pull request merged**